src/connector/src/parser/unified/upsert.rs

```diff
@@ -86,34 +77,62 @@ where
             })
         }
     }
+
+    pub fn access_field_impl(&self, desc: &SourceColumnDesc) -> super::AccessResult {
+        match desc.additional_column.column_type {
+            Some(AdditionalColumnType::Key(_)) => self.access_key(&[&desc.name], &desc.data_type),
+            None => self.access_value(&[&desc.name], &desc.data_type),
+            _ => unreachable!(),
+        }
+    }
+}
+
+/// Wraps a key-value message into an upsert event, which uses `null` value to represent `DELETE`s.
+pub struct UpsertChangeEvent<K, V>(KvEvent<K, V>);
+
+impl<K, V> Default for UpsertChangeEvent<K, V> {
+    fn default() -> Self {
+        Self(KvEvent::default())
+    }
+}
+
+impl<K, V> Deref for UpsertChangeEvent<K, V> {
+    type Target = KvEvent<K, V>;
+
+    fn deref(&self) -> &Self::Target {
+        &self.0
+    }
+}
+
+impl<K, V> DerefMut for UpsertChangeEvent<K, V> {
+    fn deref_mut(&mut self) -> &mut Self::Target {
+        &mut self.0
+    }
 }
 
 impl<K, V> ChangeEvent for UpsertChangeEvent<K, V>
 where
     K: Access,
-    V: Access,
+    V: NullableAccess,
 {
     fn op(&self) -> std::result::Result<ChangeEventOperation, AccessError> {
-        if let Ok(Some(_)) = self.access_value(&[], None) {
-            Ok(ChangeEventOperation::Upsert)
```

Comment:
Previously this is the only place where `None` is used, so it doesn't make any sense.

Reply:
The new method is smart and efficient.